TESTS/network/wifi/get_interface.cpp

```diff
@@ -0,0 +1,21 @@
+#include "mbed.h"
+#include "ESP8266Interface.h"
+
+WiFiInterface *get_interface()
```

Comment:
Do you think you could use the json files introduced in #4795 for this functionality? 

The idea is that you have a [JSON file](https://github.com/ARMmbed/mbed-os/pull/4795/files#diff-2d17f32f40dad48d9c02b70e17306d39) describing how to instantiate the interface. That file is mapped to from [some keywords](https://github.com/ARMmbed/mbed-os/pull/4795/files#diff-e03ebc83772bec28d1c4e73f9f215780). And [each target has keywords it supports](https://github.com/ARMmbed/mbed-os/pull/4795/files#diff-11fa4f2ecccaa0152f7a30e6fcb0c7af). 

Here is an [example](https://github.com/ARMmbed/mbed-os/pull/4795/files#diff-600180a025e05431787ff7d86ad01bb5R63) of how the test configuration files are used in a test.

If the tests are not target specific or the interface is outside of mbed OS, then you can create a new config file outside of mbed OS and use it in the test command like. `mbed test -m [mcu] -t [toolchain] --test-config path/outside/mbed`

Reply:
I could, these two approaches are doing the same effect.

I can rebase this PR and take your changes once you are in master.

Reply:
Actually, now after review I noticed that the `--test-config` overrides the `--app-config` so I cannot use it.
We need app-config to deliver parameters like Wifi passwords, etc, for test cases.